Libs/Requirement/RequirementFactory.cs

```diff
@@ -90,54 +90,85 @@ public Requirement GetRequirement(string name, string requirement)
+                    {  "Frostbite", ()=> playerReader.Debuffs.Rip },
+
+                    //  Paladin
+                    {  "Seal", ()=> playerReader.Buffs.Seal },
+                    {  "Aura", ()=>playerReader.Buffs.Aura },
+                    {  "Devotion Aura", ()=>playerReader.Buffs.Aura },
+                    {  "Blessing", ()=> playerReader.Buffs.Blessing },
+                    {  "Blessing of Might", ()=> playerReader.Buffs.Blessing },
+                    //-------------------------
+
+                    // Priest
+                    {  "Fortitude", ()=> playerReader.Buffs.Fortitude },
+                    {  "InnerFire", ()=> playerReader.Buffs.InnerFire },
+                    {  "Divine Spirit", ()=> playerReader.Buffs.DivineSpirit },
+                    {  "Renew", ()=> playerReader.Buffs.Renew },
+                    {  "Shield", ()=> playerReader.Buffs.Shield },
+                    //-------------------------
+                    {  "Shadow Word: Pain", ()=> playerReader.Debuffs.ShadowWordPain },
+
+                    // Rogue
                     {  "Slice And Dice", ()=> playerReader.Buffs.SliceAndDice },
-                    {  "Battle Shout", ()=> playerReader.Buffs.BattleShout },
-                    {  "Demon Skin", ()=> playerReader.Buffs.DemonSkin },
+                    //-------------------------
+
+                    // Shaman
+                    
+                    //-------------------------
+
+                    // Warlock
                     {  "Demon Armor", ()=> playerReader.Buffs.DemonArmor },
                     {  "Soul Link", ()=> playerReader.Buffs.SoulLink },
                     {  "Soulstone Resurrection", ()=> playerReader.Buffs.SoulstoneResurrection },
-                    {  "Has Pet", ()=> playerReader.PlayerBitValues.HasPet },
-
-                    {  "Demoralizing Roar", ()=> playerReader.Debuffs.Roar },
-                    {  "Faerie Fire", ()=> playerReader.Debuffs.FaerieFire },
-                    {  "Rip", ()=> playerReader.Debuffs.Rip },
+                    {  "Shadow Trance", ()=> playerReader.Buffs.ShadowTrance },
+                    //-------------------------
+                    {  "Curse of", ()=> playerReader.Debuffs.CurseOf },
+                    {  "Corruption", ()=> playerReader.Debuffs.Corruption },
+                    {  "Immolate", ()=> playerReader.Debuffs.Immolate },
+                    {  "Siphon Life", ()=> playerReader.Debuffs.SiphonLife },
+                    
+                    // Warrior
+                    {  "Battle Shout", ()=> playerReader.Buffs.BattleShout },
+                    //-------------------------
                     {  "Rend", ()=> playerReader.Debuffs.Rend },
 
-                    {  "Shadow Word: Pain", ()=> playerReader.Debuffs.ShadowWordPain },
 
-                    {  "Curse of Weakness", ()=> playerReader.Debuffs.CurseofWeakness },
-                    {  "Curse of Agony", ()=> playerReader.Debuffs.CurseofAgony },
```

Comment:
Pretty much similar thing happened here as `Demon Skin`.

The `Curse of Agony` requirement used almost all of the existing warlock profiles.
Which gonna throw missing requirement of `Curse of Agony`.

There's atleast two option here imho:
1. For better readability for the profile users create an entry for each curse which points to the `playerReader.Debuffs.CurseOf` like `"Curse of xyz", ()=> playerReader.Debuffs.CurseOf` in the long run can avoid many misunderstanding.
2. With the supplied change in this PR, please update all the available and working profiles so they wont be broken after the PR is merged, thanks.